TreeDB/docs/evidence/vector-partition-revalidation-4093/tools/run_revalidation.py

```
@@ -0,0 +1,343 @@
+    descriptor = load(Path(str(provenance["m3_database_directory"])) / "vector_partition_variant_v1.json")
+    fixture = load(Path(str(provenance["dataset_directory"])) / "fixture_manifest.json")
+    artifacts = list((root / "250k/graph-overlap-020-out").glob(
+        f"vector_partition_{str(provenance['m3_artifact_sha256'])[:12]}_*.json"))
+    if (len(artifacts) != 1 or base.sha256(artifacts[0]) != provenance["m3_artifact_sha256"] or
+            descriptor.get("base_sha") != provenance["base_sha"] or
+            descriptor.get("head_sha") != provenance["source_head"] or
+            descriptor.get("executable_sha256") != provenance["binary_sha256"] or
+            descriptor.get("artifact_sha256") != provenance["m3_artifact_sha256"] or
+            descriptor.get("fixture_checksum") != fixture.get("checksum") or
+            fixture.get("checksum") != provenance["fixture_checksum"]):
+        raise RuntimeError("M3 execution provenance changed")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)